specter-registry/src/main/java/dev/spiritstudios/specter/impl/registry/attachment/data/AttachmentMap.java

```diff
@@ -14,69 +14,56 @@
 
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.util.Collections;
 import java.util.Map;
 
 public class AttachmentMap<R, V> {
+
 	private final Registry<R> registry;
 	private final Attachment<R, V> attachment;
-	private final Map<Identifier, Object> values;
+	private final Object2ObjectOpenHashMap<Identifier, V> values;
```

Comment:
Just a minor style thing but maps should always be typed as `Map`, instead of the type of map they are.

Reply:
This is not a stylistic choice, we can't take advantage of object2object's trim otherwise

Reply:
Ah! I had no idea that function existed. Sorry